# HG changeset patch
# User jwe
# Date 745568764 0
#      Tue Aug 17 06:26:04 1993 +0000
# Node ID 74d6f5fe70a1a311a12d38e0df537db8bd0c527c
# Parent  1feceb2a86f972d66bf8f85e27c8da9f49ccf168
[project @ 1993-08-17 06:26:04 by jwe]
(process_printf_format): Use NINT for * modifier argument.

diff --git a/src/file-io.cc b/src/file-io.cc
--- a/src/file-io.cc
+++ b/src/file-io.cc
@@ -606,17 +606,17 @@ process_printf_format (char *s, tree_con
 
       if (args[fmt_arg_count].const_type ()
 	  != tree_constant_rep::scalar_constant)
 	{
 	  message (type, "`*' must be replaced by an integer");
 	  return -1;
 	}
 
-      fmt << (int) (args[fmt_arg_count++].double_value ());
+      fmt << NINT (args[fmt_arg_count++].double_value ());
       s++;
       chars_from_fmt_str++;
     }
   else
     {
       while (*s != '\0' && isdigit (*s))
 	{
 	  chars_from_fmt_str++;
@@ -646,17 +646,17 @@ process_printf_format (char *s, tree_con
 
       if (args[fmt_arg_count].const_type ()
 	  != tree_constant_rep::scalar_constant)
 	{
 	  message (type, "`*' must be replaced by an integer");
 	  return -1;
 	}
 
-      fmt << (int) (args[fmt_arg_count++].double_value ());
+      fmt << NINT (args[fmt_arg_count++].double_value ());
       s++;
       chars_from_fmt_str++;
     }
   else
     {
       while (*s != '\0' && isdigit (*s))
 	{
 	  chars_from_fmt_str++;
